xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp

```diff
@@ -156,6 +157,32 @@ bool CRendererMediaCodecSurface::RenderUpdateVideoHook(bool clear, DWORD flags,
     mci->RenderUpdate(srcRect, dstRect);
   }
 
+  /** @todo This is a really bad hack and not meant to be
+  * a solution. This needs proper fixing when android maintainer
+  * is back. Real solution: Block depending on next vblank,
+  * which is estimated here by assuming 2 buffers and frametime.
+  */
+  uint64_t sleep_time = 10; // 10 ms by default
+  double fps = g_graphicsContext.GetFPS();
+
+  if (fps > 0.0)
+  {
+    uint64_t now = XbmcThreads::SystemClockMillis();
```

Comment:
See the std::min in the fixup commits which adresses the issue.
Am 14.07.2016 8:01 vorm. schrieb "Rainer Hochecker" <
notifications@github.com>:

> In
> xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp
> <https://github.com/xbmc/xbmc/pull/10096#discussion_r70752416>:
>
> > @@ -156,6 +157,32 @@ bool CRendererMediaCodecSurface::RenderUpdateVideoHook(bool clear, DWORD flags,
> >      mci->RenderUpdate(srcRect, dstRect);
> >    }
> >
> > +  /** @todo This is a really bad hack and not meant to be
> > +  * a solution. This needs proper fixing when android maintainer
> > +  * is back. Real solution: Block depending on next vblank,
> > +  * which is estimated here by assuming 2 buffers and frametime.
> > +  */
> > +  uint64_t sleep_time = 10; // 10 ms by default
> > +  double fps = g_graphicsContext.GetFPS();
> > +
> > +  if (fps > 0.0)
> > +  {
> > +    uint64_t now = XbmcThreads::SystemClockMillis();
>
> this is not correct. SystemClockMillis is of type unsigned int and can
> wrap. If you don't take this into account, you may sleep here for a very
> long time.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/xbmc/xbmc/pull/10096/files/96837cc696800d0968a92a7e7b869a5d8531e530#r70752416>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABCfHR0KJwLGOF2LvGcszVa0sMkXo8uyks5qVdCjgaJpZM4JIq8y>
> .
>